### PR TITLE
fix: Tag changes do not cause updates to firehose destinations

### LINKED ISSUE #25396: Updating tags for aws_kinesis_firehose_delivery_stream also updates the Redshift delivery settings
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

- Terraform CLI version  1.2.3
- Terraform AWS Provider Version 4.18.0

### Affected Resource(s)

* aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_kinesis_firehose_delivery_stream" "example" {
  name        = "example_stream"
  destination = var.kinesis_destination_configuration

  s3_configuration {
    role_arn           = aws_iam_role.role.arn
    bucket_arn         = data.aws_s3_bucket.bucket.arn
    prefix             = var.s3_path
    buffer_size        = var.kinesis_buffer_size
    compression_format = "GZIP"
  }

  redshift_configuration {
    role_arn        = aws_iam_role.role.arn
    cluster_jdbcurl = local.cluster_jdbcurl

    username = "username"
    password = "password"

    data_table_name    = var.data_table_name
    copy_options       = var.data_copy_options
    data_table_columns = join(", ", var.data_table_columns)

    processing_configuration {
      enabled = "false"
    }
    cloudwatch_logging_options {
      enabled         = true
      log_group_name  = aws_cloudwatch_log_group.log_group.name
      log_stream_name = aws_cloudwatch_log_stream.redshift.name
    }
  }

  lifecycle {
    ignore_changes = [
      redshift_configuration.0.username,
      redshift_configuration.0.password,
    ]
  }
}
```

### Expected Behavior

Terraform should just apply the tags without making any other changes.

### Actual Behavior

Tags were applied but the password for the `redshift_configuration` was blanked out, so the stream started to fail to deliver any content to Redshift.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

We set the username and password to placeholder values that we then change manually in the AWS Management Console. We do this to keep credentials out of Terraform state, where it can be extracted. We believe that when the Terraform AWS Provider goes to update tags, it also goes to update delivery settings, which then sets the credentials to the placeholder values.

### References

* #0000

### PR DESCRIPTION
**Pull request by mbamber**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #25396

Output from acceptance testing:

```
$ AWS_DEFAULT_REGION=eu-west-1 AWS_SDK_LOAD_CONFIG=1 AWS_PROFILE=dev make testacc TESTS=TestAccFirehoseDeliveryStream_tagUpdates PKG=firehose
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/firehose/... -v -count 1 -parallel 20 -run='TestAccFirehoseDeliveryStream_tagUpdates'  -timeout 180m
=== RUN   TestAccFirehoseDeliveryStream_tagUpdates
=== PAUSE TestAccFirehoseDeliveryStream_tagUpdates
=== CONT  TestAccFirehoseDeliveryStream_tagUpdates
--- PASS: TestAccFirehoseDeliveryStream_tagUpdates (375.03s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/firehose	377.636s
```

I don't really know how to test that this fixes the underlying issue, as the redshift password is not returned in API requests (and as such it's not possible to check that it hasn't been changed, AFAIK). But from comments on the ticket this appears to be the correct solution. Any advice here greatly appreciated.

### CODE REVIEW

**Comment by mbamber** [comment]
> Your current settings do not allow maintainers to make such edits

I can't see a setting to change this anywhere...

**Comment by ewbankkit** [comment]
@mbamber I think that's because the PR has been submitted from a fork in an organization's GitHub account, not a personal account; That's to be expected.

File: .changelog/26451.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/aws_kinesis_firehose_delivery_stream: Updating tags no longer causes an uncessary update
```
Comment (ewbankkit) [inline .changelog/26451.txt:2]: Fix spelling.

```suggestion
resource/aws_kinesis_firehose_delivery_stream: Updating `tags` no longer causes an unnecessary update
```

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccFirehoseDeliveryStream_' PKG=firehose ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/firehose/... -v -count 1 -parallel 2  -run=TestAccFirehoseDeliveryStream_ -timeout 180m
=== RUN   TestAccFirehoseDeliveryStream_basic
=== PAUSE TestAccFirehoseDeliveryStream_basic
=== RUN   TestAccFirehoseDeliveryStream_disappears
=== PAUSE TestAccFirehoseDeliveryStream_disappears
=== RUN   TestAccFirehoseDeliveryStream_s3basic
=== PAUSE TestAccFirehoseDeliveryStream_s3basic
=== RUN   TestAccFirehoseDeliveryStream_s3basicWithPrefixes
=== PAUSE TestAccFirehoseDeliveryStream_s3basicWithPrefixes
=== RUN   TestAccFirehoseDeliveryStream_s3basicWithSSE
=== PAUSE TestAccFirehoseDeliveryStream_s3basicWithSSE
=== RUN   TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyARN
=== PAUSE TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyARN
=== RUN   TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyType
=== PAUSE TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyType
=== RUN   TestAccFirehoseDeliveryStream_s3basicWithTags
=== PAUSE TestAccFirehoseDeliveryStream_s3basicWithTags
=== RUN   TestAccFirehoseDeliveryStream_s3KinesisStreamSource
=== PAUSE TestAccFirehoseDeliveryStream_s3KinesisStreamSource
=== RUN   TestAccFirehoseDeliveryStream_s3WithCloudWatchLogging
=== PAUSE TestAccFirehoseDeliveryStream_s3WithCloudWatchLogging
=== RUN   TestAccFirehoseDeliveryStream_s3Updates
=== PAUSE TestAccFirehoseDeliveryStream_s3Updates
=== RUN   TestAccFirehoseDeliveryStream_extendedS3basic
=== PAUSE TestAccFirehoseDeliveryStream_extendedS3basic
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversion_enabled
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversion_enabled
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3_externalUpdate
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3_externalUpdate
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionDeserializer_update
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionDeserializer_update
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionHiveJSONSerDe_empty
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionHiveJSONSerDe_empty
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOpenXJSONSerDe_empty
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOpenXJSONSerDe_empty
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOrcSerDe_empty
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOrcSerDe_empty
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionParquetSerDe_empty
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionParquetSerDe_empty
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionSerializer_update
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionSerializer_update
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3_errorOutputPrefix
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3_errorOutputPrefix
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3_S3BackupConfiguration_ErrorOutputPrefix
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3_S3BackupConfiguration_ErrorOutputPrefix
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3Processing_empty
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3Processing_empty
=== RUN   TestAccFirehoseDeliveryStream_extendedS3KMSKeyARN
=== PAUSE TestAccFirehoseDeliveryStream_extendedS3KMSKeyARN
=== RUN   TestAccFirehoseDeliveryStream_extendedS3DynamicPartitioning
=== PAUSE TestAccFirehoseDeliveryStream_extendedS3DynamicPartitioning
=== RUN   TestAccFirehoseDeliveryStream_extendedS3Updates
=== PAUSE TestAccFirehoseDeliveryStream_extendedS3Updates
=== RUN   TestAccFirehoseDeliveryStream_ExtendedS3_kinesisStreamSource
=== PAUSE TestAccFirehoseDeliveryStream_ExtendedS3_kinesisStreamSource
=== RUN   TestAccFirehoseDeliveryStream_redshiftUpdates
=== PAUSE TestAccFirehoseDeliveryStream_redshiftUpdates
=== RUN   TestAccFirehoseDeliveryStream_tagUpdates
=== PAUSE TestAccFirehoseDeliveryStream_tagUpdates
=== RUN   TestAccFirehoseDeliveryStream_splunkUpdates
=== PAUSE TestAccFirehoseDeliveryStream_splunkUpdates
=== RUN   TestAccFirehoseDeliveryStream_Splunk_ErrorOutputPrefix
=== PAUSE TestAccFirehoseDeliveryStream_Splunk_ErrorOutputPrefix
=== RUN   TestAccFirehoseDeliveryStream_httpEndpoint
=== PAUSE TestAccFirehoseDeliveryStream_httpEndpoint
=== RUN   TestAccFirehoseDeliveryStream_HTTPEndpoint_ErrorOutputPrefix
=== PAUSE TestAccFirehoseDeliveryStream_HTTPEndpoint_ErrorOutputPrefix
=== RUN   TestAccFirehoseDeliveryStream_HTTPEndpoint_retryDuration
=== PAUSE TestAccFirehoseDeliveryStream_HTTPEndpoint_retryDuration
=== RUN   TestAccFirehoseDeliveryStream_elasticSearchUpdates
=== PAUSE TestAccFirehoseDeliveryStream_elasticSearchUpdates
=== RUN   TestAccFirehoseDeliveryStream_elasticSearchEndpointUpdates
=== PAUSE TestAccFirehoseDeliveryStream_elasticSearchEndpointUpdates
=== RUN   TestAccFirehoseDeliveryStream_elasticSearchWithVPCUpdates
=== PAUSE TestAccFirehoseDeliveryStream_elasticSearchWithVPCUpdates
=== RUN   TestAccFirehoseDeliveryStream_Elasticsearch_ErrorOutputPrefix
=== PAUSE TestAccFirehoseDeliveryStream_Elasticsearch_ErrorOutputPrefix
=== RUN   TestAccFirehoseDeliveryStream_missingProcessing
=== PAUSE TestAccFirehoseDeliveryStream_missingProcessing
=== CONT  TestAccFirehoseDeliveryStream_basic
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3_errorOutputPrefix
--- PASS: TestAccFirehoseDeliveryStream_basic (97.79s)
=== CONT  TestAccFirehoseDeliveryStream_s3Updates
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3_errorOutputPrefix (159.23s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionSerializer_update
--- PASS: TestAccFirehoseDeliveryStream_s3Updates (154.91s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionParquetSerDe_empty
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionSerializer_update (224.90s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOrcSerDe_empty
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionParquetSerDe_empty (133.67s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOpenXJSONSerDe_empty
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOpenXJSONSerDe_empty (113.61s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionHiveJSONSerDe_empty
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionOrcSerDe_empty (131.43s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionDeserializer_update
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionHiveJSONSerDe_empty (112.40s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3_externalUpdate
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversionDeserializer_update (131.83s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversion_enabled
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3_externalUpdate (113.23s)
=== CONT  TestAccFirehoseDeliveryStream_extendedS3basic
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3DataFormatConversion_enabled (143.78s)
=== CONT  TestAccFirehoseDeliveryStream_Splunk_ErrorOutputPrefix
--- PASS: TestAccFirehoseDeliveryStream_extendedS3basic (174.20s)
=== CONT  TestAccFirehoseDeliveryStream_missingProcessing
--- PASS: TestAccFirehoseDeliveryStream_Splunk_ErrorOutputPrefix (147.19s)
=== CONT  TestAccFirehoseDeliveryStream_Elasticsearch_ErrorOutputPrefix
--- PASS: TestAccFirehoseDeliveryStream_missingProcessing (144.51s)
=== CONT  TestAccFirehoseDeliveryStream_elasticSearchWithVPCUpdates
=== CONT  TestAccFirehoseDeliveryStream_Elasticsearch_ErrorOutputPrefix
    delivery_stream_test.go:1565: Step 1/5 error: Error running apply: exit status 1
        
        Error: error waiting for Kinesis Firehose Delivery Stream (tf-acc-test-1009201887386327658) create: timeout while waiting for state to become 'ACTIVE' (last state: 'CREATING', timeout: 20m0s)
        
          with aws_kinesis_firehose_delivery_stream.test,
          on terraform_plugin_test.tf line 129, in resource "aws_kinesis_firehose_delivery_stream" "test":
         129: resource "aws_kinesis_firehose_delivery_stream" "test" {
        
--- FAIL: TestAccFirehoseDeliveryStream_Elasticsearch_ErrorOutputPrefix (2455.31s)
=== CONT  TestAccFirehoseDeliveryStream_elasticSearchEndpointUpdates
--- PASS: TestAccFirehoseDeliveryStream_elasticSearchWithVPCUpdates (3181.78s)
=== CONT  TestAccFirehoseDeliveryStream_elasticSearchUpdates
--- PASS: TestAccFirehoseDeliveryStream_elasticSearchEndpointUpdates (1402.76s)
=== CONT  TestAccFirehoseDeliveryStream_HTTPEndpoint_retryDuration
--- PASS: TestAccFirehoseDeliveryStream_HTTPEndpoint_retryDuration (210.15s)
=== CONT  TestAccFirehoseDeliveryStream_HTTPEndpoint_ErrorOutputPrefix
--- PASS: TestAccFirehoseDeliveryStream_HTTPEndpoint_ErrorOutputPrefix (161.22s)
=== CONT  TestAccFirehoseDeliveryStream_httpEndpoint
--- PASS: TestAccFirehoseDeliveryStream_httpEndpoint (175.61s)
=== CONT  TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyARN
--- PASS: TestAccFirehoseDeliveryStream_elasticSearchUpdates (1490.50s)
=== CONT  TestAccFirehoseDeliveryStream_s3WithCloudWatchLogging
--- PASS: TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyARN (384.00s)
=== CONT  TestAccFirehoseDeliveryStream_s3KinesisStreamSource
--- PASS: TestAccFirehoseDeliveryStream_s3KinesisStreamSource (58.43s)
=== CONT  TestAccFirehoseDeliveryStream_s3basicWithTags
--- PASS: TestAccFirehoseDeliveryStream_s3WithCloudWatchLogging (151.61s)
=== CONT  TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyType
--- PASS: TestAccFirehoseDeliveryStream_s3basicWithTags (173.94s)
=== CONT  TestAccFirehoseDeliveryStream_tagUpdates
--- PASS: TestAccFirehoseDeliveryStream_s3basicWithSSEAndKeyType (263.72s)
=== CONT  TestAccFirehoseDeliveryStream_splunkUpdates
--- PASS: TestAccFirehoseDeliveryStream_tagUpdates (383.38s)
=== CONT  TestAccFirehoseDeliveryStream_extendedS3KMSKeyARN
--- PASS: TestAccFirehoseDeliveryStream_splunkUpdates (219.95s)
=== CONT  TestAccFirehoseDeliveryStream_extendedS3Updates
--- PASS: TestAccFirehoseDeliveryStream_extendedS3KMSKeyARN (121.64s)
=== CONT  TestAccFirehoseDeliveryStream_redshiftUpdates
--- PASS: TestAccFirehoseDeliveryStream_extendedS3Updates (152.63s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3_kinesisStreamSource
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3_kinesisStreamSource (71.04s)
=== CONT  TestAccFirehoseDeliveryStream_extendedS3DynamicPartitioning
--- PASS: TestAccFirehoseDeliveryStream_extendedS3DynamicPartitioning (142.56s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3Processing_empty
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3Processing_empty (111.29s)
=== CONT  TestAccFirehoseDeliveryStream_ExtendedS3_S3BackupConfiguration_ErrorOutputPrefix
--- PASS: TestAccFirehoseDeliveryStream_redshiftUpdates (379.82s)
=== CONT  TestAccFirehoseDeliveryStream_s3basicWithPrefixes
--- PASS: TestAccFirehoseDeliveryStream_s3basicWithPrefixes (146.94s)
=== CONT  TestAccFirehoseDeliveryStream_s3basicWithSSE
--- PASS: TestAccFirehoseDeliveryStream_ExtendedS3_S3BackupConfiguration_ErrorOutputPrefix (167.36s)
=== CONT  TestAccFirehoseDeliveryStream_disappears
--- PASS: TestAccFirehoseDeliveryStream_disappears (149.73s)
=== CONT  TestAccFirehoseDeliveryStream_s3basic
--- PASS: TestAccFirehoseDeliveryStream_s3basic (103.80s)
--- PASS: TestAccFirehoseDeliveryStream_s3basicWithSSE (278.92s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/firehose	7275.161s
```

Failure is unrelated to this change.

**Comment by ewbankkit** [comment]
@mbamber Thanks for the contribution :tada: :clap:.

**Pull request merged**